etcdclient/api.go

```diff
@@ -0,0 +1,49 @@
+package etcdclient
+
+import (
+	"log"
+	"os"
+	"time"
+
+	etcd "go.etcd.io/etcd/clientv3"
+	etcdtransport "go.etcd.io/etcd/pkg/transport"
+)
+
+const (
+	certPath      = "/etc/ssl/etcd/ssl/"
+	trustedCAFile = certPath + "ca.pem"
+)
+
+// New initializes etcd config structures and returns an etcd client
+func New(endPoints []string, autoSyncInterval time.Duration, dialTimeout time.Duration) (etcdClient *etcd.Client, err error) {
+	var caFile string
+
+	// If we have a local CA then use it.  Otherwise, use the system wide CA
+	_, statErr := os.Stat(trustedCAFile)
+	if os.IsExist(statErr) {
+		caFile = trustedCAFile
+	}
+
+	// Certs are named based on the host name.
+	h, _ := os.Hostname()
+	certFile := certPath + "node-" + h + ".pem"
+	keyFile := certPath + "node-" + h + "-key.pem"
+
+	tlsInfo := etcdtransport.TLSInfo{
+		CertFile:      certFile,
+		KeyFile:       keyFile,
+		TrustedCAFile: caFile,
+	}
```

Comment:
If New() took a `*etcdtransport.TLSInfo` argument then having that value be nil could indicate a simple TCP connection is used.

but then the code that is highlighted here either 1) gets repeated in multiple locations or 2) gets moved to some other function, which somewhat negates the value of creating `etcdclient`.

it's your call what you want to do here, if anything ...